### PR TITLE
Policy search

### PR DESCRIPTION
**Pull request by Smightym8**
This pull request includes several changes to the `src/App.tsx` file, the `src/api/openAPI` directory, and the `src/components/tables/SmartMeterPoliciesTable.tsx` file. The primary focus is on adding new features, updating existing models, and improving the overall functionality of the application. The most important changes are summarized below:

### New Features:
* [`src/App.tsx`](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R13): Added `PolicySearchPage` import and route to the application. [[1]](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R13) [[2]](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R48-R51)

### API Updates:
* [`src/api/openAPI/apis/measurement-api.ts`](diffhunk://#diff-4876c829876aa66092e164b560a07fda0a49de31659a7ee5805967eb02eca03aR1-R159): Introduced `MeasurementApi` with methods for fetching measurement data.
* [`src/api/openAPI/apis/policy-api.ts`](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R25-R26): Added `searchPolicies` method to the `PolicyApi` for searching policies by max price and measurement resolution. [[1]](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R25-R26) [[2]](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R104-R145) [[3]](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R189-R201) [[4]](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R230-R239) [[5]](diffhunk://#diff-f29287179bb313986a5cc255c4987503cbad31f17bbf82e23eda56d600e84d79R271-R282)

### Model Changes:
* [`src/api/openAPI/models/measurement-raw-dto.ts`](diffhunk://#diff-b7a3eee74f4d9809a28e38606c0ee64cd2c3f482fb81051bf0b715c908a8c896R1-R114): Added new model `MeasurementRawDto`.
* [`src/api/openAPI/models/policy-dto.ts`](diffhunk://#diff-9365ecf28559a37d6c33eb91ecb48b717e186e4835c4729e1cc324da46a21bd2L34-R58): Updated `PolicyDto` to make fields `id`, `name`, `measurementResolution`, `locationResolution`, and `price` required.

### Code Cleanup:
* [`src/api/openAPI/.openapi-generator/FILES`](diffhunk://#diff-aad06550bd1e74cbc68879bd4ae6edd7a263227b0232a206deeaa5aed220c763R5-L6): Removed `policyrequest-api.ts` and added `measurement-api.ts`. [[1]](diffhunk://#diff-aad06550bd1e74cbc68879bd4ae6edd7a263227b0232a206deeaa5aed220c763R5-L6) [[2]](diffhunk://#diff-aad06550bd1e74cbc68879bd4ae6edd7a263227b0232a206deeaa5aed220c763R18-L25)
* [`src/api/openAPI/api.ts`](diffhunk://#diff-04df539f4c4e176847ef184e1ad76f40deeb0b3384a14c3ea6e48bbba9e473c7R18-L19): Updated exports to include `measurement-api` and exclude `policyrequest-api`.

### UI Improvements:
* [`src/components/tables/SmartMeterPoliciesTable.tsx`](diffhunk://#diff-586814c275b8fafcf68b1c65d25da3a5572c5923c1231c18a44daa440648f1f0L92-R101): Updated table rendering to remove null checks for policy fields and improved the no policies message.

### CODE REVIEW

**Comment by 1r-1na** [comment]
- When clicking on "Clear Filters", the filter-texts remain, but the actual filters were unset.
image

**Review by 1r-1na — DISMISSED** [review]
- When clicking on "Clear Filters", the filter-texts remain, but the actual filters were unset.
image

File: src/pages/navbar/PolicySearchPage.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        if (!price) {
+            price = undefined;
+        }
+
+        void loadPolicies(price, measurementResolution);
+    };
+
+    const handleReset = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
```
Comment (1r-1na) [inline src/pages/navbar/PolicySearchPage.tsx:56]: By removing line 56 `event.preventDefault()` mentioned problem of no clearing the filter texts, would be solved.

**Review by iiNomad23 — APPROVED** [review]
lgtm

**Pull request merged**